newton/tests/test_viewer_get_frame.py

```
@@ -112,6 +112,27 @@ def test_headless_frame_capture_across_devices(self):
         finally:
             viewer.close()
 
+    def test_headless_capture_main_image_frame(self):
+        """Verify get_frame captures a main image rendered headlessly."""
+        try:
+            viewer = newton.viewer.ViewerGL(width=64, height=48, headless=True)
+        except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)